technical-docs/designs/infra/managed-identity.md

```diff
@@ -0,0 +1,251 @@
+  Bedrock: i.e.: testing files in respective folders for template environments.
+
+## 3. Design Details
+
+This design seeks to introduce modular testing for terraform known as
+`Test Fixtures` based on best practices initially introduced by
+[Project Cobalt](github.com/microsoft/cobalt). The test fixtures decouples
+terraform commands to respective pipeline templats to be called and dynamically
+populated by a targeted template test.
+
+### 3.1 Embed new Infrastructure DevOps Model Flow - Continuous Integration
+
+Bedrock infrastructure integration tests have problematic gaps that do not
+account for terraform unit testing, state validation to live environments and
+staged release management for Bedrock versioning. In this design we wish to
+provide a single template leveraging MSI that verifies a new Infrastructure
+Testing Workflow that improves on the current Bedrock test harness.
+
+This design is intended to address expected core testing functionality
+including:
+
+- Support deployment of application-hosting infrastructure that will eventually
+  house the actual application service components Capture basic metrics and
+  telemetry from the deployment process for monitoring of ongoing pipeline
+  performance and diagnosis of any deployment failures
+- Support deployment into multiple staging environments
+- Execute automated unit-level and integration-level tests against the
+  resources, prior to deployment into any long-living environments
+- Provide a manual approval process to gate deployment into long-living
+  environments
+- Provide detection, abort, and reporting of deployment status when a failure
+  occurs.
+
+![](infratestflow.png)
+
+The proposed new Infrastructure Devops Flow for Terraform Testing can be
+separated by 4 key functionalities:
+
+1. Test Suite Initialization - Provisioning global artifacts, secrets and
+   dependencies needed for targeted whitelisted test matrix.
+2. Static Validation - Environment initialization, code validation, inspection,
+   terraform security compliance, and terraform module unit tests.
+3. Dynamic Validation - Targeted environment interoperability, integration
+   tests, cloud deployment, de-provisioning of resources, error reporting.
+4. QA- Peer approval, release management, feature staging, acceptance test
+   within live cluster.
+
+> The diagram above contains green check marks that indicate preexisting Bedrock
+> testing components that are already implemented through the current test
+> harness.
+
+### 3.2 Creation of Managed Identity enable AKS Gitops Environments
+
+A new AKS Bedrock template with Managed Identity enabled, (`azure-MI`), will be
+added the the collection of sample environments. This template will be an
+upgraded derivative of the `azure-simple` template, with a new dependency on
+`azure-common-infra` and will contain the following:
+
+- Managed Identity System Level for AKS
+- Pod Identity Security Policy
+- Backend State
```

Comment:
Is the plan to use remote backend state for both local and ADO test runs?

Reply:
Yes, if local then the state file will be written to disk.

Reply:
There may be a need to replicate modules in the case of MI vs SP.  I don't know the scope off the top of my head.  The changes should isolate any role/identity stuff into the equivalent of azure-common-infra when possible